R/scenario_helpers.R

```diff
@@ -0,0 +1,387 @@
+    z <- z[compartment == "hospitalised",
+      list(value = sum(value)),
+      by = "time"
+    ]
+    z$measure <- "total_hosp"
+
+    z
+  })
+
+  dt_list <- Map(
+    df_list, total_hosp_list, names,
+    f = function(incidence_df, hosp_df, n) {
+      dt <- data.table::rbindlist(
+        list(incidence_df, hosp_df)
+      )
+      dt$tag <- n
+
+      dt
+    }
+  )
+
+  dt <- data.table::rbindlist(dt_list)
+
+  dt
+}
+
+#' Get details of economic costs from model outputs
+#'
+#' @param l A list of `<daedalus.output>` objects.
+#'
+#' @keywords internal
+get_econ_costs_list <- function(l) {
+  # no input checks on internal functions
+  econ_costs_list <- lapply(l, function(x) {
+    z <- daedalus::get_costs(x)
+    z[["economic_costs"]][
+      c("economic_cost_closures", "economic_cost_absences")
+    ]
+  })
+
+  econ_costs_list <- lapply(econ_costs_list, as.data.frame)
+
+  data.table::rbindlist(econ_costs_list)
+}
+
+#' Run multiple DAEDALUS scenarios
+#'
+#' @inheritParams daedalus::daedalus_rtm
+#'
+#' @param duration A vector of integer-ish numbers giving the durations over
+#' which to run scenarios. Each scenario is run for each duration.
+#'
+#' @export
+run_scenarios <- function(country,
+                          infection,
+                          response_strategy = "none",
+                          response_time_start = 0,
+                          response_time_end = 0,
+                          duration = 100) {
+  # input checking
+  checkmate::assert_multi_class(country, c("daedalus_country", "character"))
+  if (is.character(country)) {
+    country <- daedalus::daedalus_country(country)
+  }
+  checkmate::assert_multi_class(
+    infection, c("daedalus_infection", "character", "list")
+  )
+
+  checkmate::assert_multi_class(
+    response_strategy, c("character", "list")
+  )
+
+  # check simple args
+  checkmate::assert_number(
+    response_time_start,
+    lower = 0, finite = TRUE
+  )
+  checkmate::assert_number(
+    response_time_end,
+    lower = 0, finite = TRUE
+  )
+  checkmate::assert_numeric(
+    duration,
+    lower = 0, finite = TRUE
+  )
+
+
+  # handle custom and pre-defined response scenarios
+  if (is.list(response_strategy)) {
+    resp_predef <- unlist(Filter(is.character, response_strategy))
+    checkmate::assert_subset(resp_predef, names(daedalus::closure_data))
+    custom_responses <- Filter(is.numeric, response_strategy)
+    resp_names_custom <- names(custom_responses)
+
+    if (is.null(resp_names_custom)) {
+      resp_names_custom <- glue::glue(
+        "custom_response_{seq_along(custom_responses)}"
+      )
+    }
+  }
+  scenarios <- data.table::CJ(
+    response = response_strategy, duration = duration, sorted = FALSE
+  )
+
+  # get range names from disease_x or synthetic names
+  # assuming 4-digit list lengths
+  disease_tags <- names(infection)
+  if (is.null(disease_tags)) {
+    disease_tags <- sprintf("replicate_%04i", seq_along(infection))
```

Comment:
The other way of achieving this would be to use

```
formatC(seq_along(infection), flag = "0", width = floor(log10(length(infection))) + 1)
```

to zero-pad numbers up to a fixed width within the group

Reply:
I think with recent changes to {daedalus} this could go away in a follow-up PR, I'll check if I can remove this.